arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx

```diff
@@ -266,7 +267,123 @@ Note the `logs` field within the transaction receipt:
       ]
 ```
 
-From the `logs` field, we can see the `topics[0]` is `0xa937..cfa`, which is the event signature of `Hi(address)`. You can verify this using [4 byte directory](https://www.4byte.directory/event-signatures/?bytes_signature=0xa9378d5bd800fae4d5b8d4c6712b2b64e8ecc86fdc831cb51944000fc7c8ecfa) - `topics[1]` is `Your_address`, which is exactly what we defined above!
+## Option 4: Customize gas usage for a specific method
+
+In the above, we introduced how to define a new precompile function. However, if it is simply defined without performing gas collection within the function, it will lead to potential DOS attacks. For this reason, we will continue to introduce how to perform gas collection in precompile. (event does not need to customize the gas cost, because the program will calculate the gas cost of the event when it starts).
+
+At the same time, you can also customize a calculation library with lower gas usage than conventional solidity chain contracts based on this document, so that users can use it as a third-party library call with low gas usage.
```

Comment:
```suggestion
In addition to introducing gas costs where they don't exist, you can also customize gas costs where they're already being incurred. To demonstrate this approach, consider the `GetBalance` method in `ArbInfo.go`:
```

Question: We're using `GetBalance` as an example, and it seems like we're implementing a sort of "override" (in quotes because it's a different method name, but same signature and purpose) that still costs gas, but just customizes the amount of gas that it costs.

If my understanding is correct, the introduction to this section suggests that you want to "incur a gas cost where it isn't yet being incurred" as a mitigation against DOS attacks - but in this example we're "changing the amount of gas where a cost is already being incurred".

My edits are under the assumption that we want to provide guidance for _both_ scenarios - 1) introduce new gas costs to render DOS attacks infeasible, and 2) customize gas costs wherever they're already being incurred. Is this assumption correct? @Jason-W123 @TucksonDev 🙏

Reply:
Thanks @symbolpunk , I made this to `GetBalance` is because it would be easier to do that. Follow your thought, if we use `SayHi` will better?

Reply:
@Jason-W123 I think `GetBalance` is a great example because it represents more of a real-world situation. I'm trying to confirm that we want to demonstrate both scenarios:

1. Create new gas costs
2. Update existing gas costs 

As long as my suggested edits (including the below edits) align with your original intent, I think we're on the same page 🙏

Reply:
Yes, we just want to demonstrate how to add gas cost to a precompile method, so I use a existing method as example, so it would be easy for devs to understand